hudi-common/src/main/java/org/apache/hudi/common/model/HoodieAvroRecord.java

```diff
@@ -168,6 +182,8 @@ public boolean isDelete(Schema recordSchema, Properties props) throws IOExceptio
     }
     if (this.data instanceof BaseAvroPayload) {
       return ((BaseAvroPayload) this.data).isDeleted(recordSchema, props);
+    } else if (this.data instanceof HoodieMetadataPayload) {
```

Comment:
I feel we should not leak the payload into to this class. 
as of now, we only use `BaseAvroPayload` and no other payloads in here. 

I checked the impl of 
```
  public boolean isDeleted() {
    return isDeletedRecord;
  }
```
in HoodieMetadataPayload. and wondering why can't we just impl the same method. 

```
  public boolean isDeleted(Schema schema, Properties props) {
    return isDeletedRecord;
  }
```

and it should work right?

Reply:
Are you suggesting adding a new method to the interface for payloads?

Reply:
If `HoodieMetadataPayload` and `BaseAvroPayload` share the same interface, then we can simplify the code.

Reply:
I think we can unify the `#isDelete` interface while we made a deep refactoring for `HoodieMetadataPayload` in the future, the current change is not a blocker and looks good to me, maybe we fire a JIRA task first to mark a TODO there.

Reply:
sg then.